dev/attest/src/fs.ts

```diff
@@ -187,3 +163,53 @@ export type SourceFileEntry = [path: string, contents: string]
 
 export const getSourceFileEntries = (dir = "."): SourceFileEntry[] =>
     getSourceFilePaths(dir).map((path) => [path, readFile(path)])
+
+export type ReplacementDictionary = Record<
+    string,
+    { pattern: RegExp; replacement: `"${string}"` }
+>
+export type Matchers = ReadonlyArray<Matcher>
+export type Matcher = {
+    pattern: RegExp
+    replacement: string
+}
+
+/**
+ * Given a set of {@link Matchers} and an input string,
+ * replace all will replace
```

Comment:
Looks like you missed the end of this description! As you have noticed, JSDoc is not mandatory for internal tooling like this, but it is definitely appreciated 😊